test/test_rhat.py

```diff
@@ -33,11 +41,117 @@ def test_rhat_ragged():
-        chain1 = [1.01, 1.2, 1.3, 1.4]
-        chains = [chain1]
-        rhat(chains)
+def test_rhat_size_exceptions():
+    bad1 = []
+    bad2 = ([[1.01, 1.2, 1.3, 1.4]],)
+    bad3 = [[1, 2, 3], [4], [5, 6, 7, 8, 9]]
+    for chains in [bad1, bad2, bad3]:
+        with pt.raises(ValueError):
+            rhat(chains)
+
+
+def test_split_chains():
+    np.testing.assert_equal([], split_chains([]))
+    np.testing.assert_equal([[1], []], split_chains([[1]]))
+    np.testing.assert_equal([[1], [2]], split_chains([[1, 2]]))
+    np.testing.assert_equal([[1, 2], [3]], split_chains([[1, 2, 3]]))
+    np.testing.assert_equal([[1, 2, 3], [4, 5, 6]], split_chains([[1, 2, 3, 4, 5, 6]]))
+    np.testing.assert_equal(
+        [[1, 2], [3], [4, 5], [6, 7]], split_chains([[1, 2, 3], [4, 5, 6, 7]])
+    )
+
+
+def test_split_rhat():
+    np.testing.assert_allclose(rhat([[1, 2], [3, 4]]), split_rhat([[1, 2, 3, 4]]))
+    np.testing.assert_allclose(
+        rhat([[1, 2, 2], [3, 4, 3]]), split_rhat([[1, 2, 2, 3, 4, 3]])
+    )
+    np.testing.assert_allclose(
+        rhat([[1, -2, 3], [4, 5, 6], [7, 8], [9, 12]]),
+        split_rhat([[1, -2, 3, 4, 5, 6], [7, 8, 9, 12]]),
+    )
+
+
+def test_split_rhat_size_exceptions():
+    bad1 = []
+    bad2 = [[1.01, 1.2, 1.3]]
+    bad3 = [[1, 2, 3], [4, 5, 6, 7]]
+    for chains in [bad1, bad2, bad3]:
+        with pt.raises(ValueError):
+            split_rhat(chains)
+
+
+def test_rank_chains():
+    np.testing.assert_equal([], rank_chains([]))
+    np.testing.assert_equal([[1]], rank_chains([[2.3]]))
+    np.testing.assert_equal([[1, 2]], rank_chains([[2.3, 4.9]]))
+    np.testing.assert_equal([[2, 3, 1]], rank_chains([[3.9, 5.2, 2.1]]))
+    np.testing.assert_equal([[2], [1]], rank_chains([[4.2], [1.9]]))
+    np.testing.assert_equal([[2, 3], [1, 4]], rank_chains([[4.2, 5.7], [1.9, 12.2]]))
+    np.testing.assert_equal(
+        [[2, 3], [5, 4], [1, 6]], rank_chains([[4.2, 5.7], [7.2, 6.1], [-12.9, 107]])
+    )
+
+
+def rank_norm(r, S):
+    return sp.stats.norm.ppf((r - 0.325) / (S - 0.25))
+
+
+def test_rank_normalize_chains():
+    np.testing.assert_equal([], rank_normalize_chains([]))
+
+    rn11 = rank_norm(1, 1)
+    np.testing.assert_allclose([[rn11]], rank_normalize_chains([[32.7]]))
+    np.testing.assert_allclose([[rn11]], rank_normalize_chains([[-10.7]]))
+
+    rn12 = rank_norm(1, 2)
+    rn22 = rank_norm(2, 2)
+    np.testing.assert_allclose([[rn22, rn12]], rank_normalize_chains([[3.9, 1.8]]))
+
+    rn13 = rank_norm(1, 3)
+    rn23 = rank_norm(2, 3)
+    rn33 = rank_norm(3, 3)
+    np.testing.assert_allclose(
+        [[rn23, rn33, rn13]], rank_normalize_chains([[3.5, 5.9, 1.0]])
+    )
+
+    rn14 = rank_norm(1, 4)
+    rn24 = rank_norm(2, 4)
+    rn34 = rank_norm(3, 4)
+    rn44 = rank_norm(4, 4)
+    np.testing.assert_allclose(
+        [[rn34, rn24], [rn14, rn44]], rank_normalize_chains([[3.9, 3.1], [2.2, 5.9]])
+    )
+
+
+def test_rank_normalized_rhat_size_exceptions():
```

Comment:
Same as above

Reply:
also done.

Reply:
@bob-carpenter this code also did not get updated/pushed it seems